databridge.examples/databridge.examples.aas-jsonata-mqtt/src/test/java/org/eclipse/digitaltwin/basyx/databridge/examples/aasjsonatamqtt/test/TestAASUpdater.java

```diff
@@ -0,0 +1,265 @@
+import io.moquette.broker.config.ResourceLoaderConfig;
+
+/**
+ * A test of aas-jsonata-mqtt 
+ * @author rana
+ *
+ */
+
+public class TestAASUpdater {
+
+	private static Logger logger = LoggerFactory.getLogger(TestAASUpdater.class);
+
+	private static AASServerComponent aasServer;
+	protected static IIdentifier deviceAASPlainId = new CustomId("TestUpdatedDeviceAAS");
+
+	private static BaSyxContextConfiguration aasContextConfig;
+	private static InMemoryRegistry registry = new InMemoryRegistry();
+	
+	protected static Server mqttBroker;
+	private static DataBridgeComponent updater;
+	private static String mqtt_broker_url = "tcp://broker.mqttdashboard.com:1883";
+	private static String user_name = "test1";
+	private static String password = "1234567";
+	private static String client_id = UUID.randomUUID().toString();
+	protected static String receivedMessage;
+	
+	
+	@BeforeClass
+	public static void setUp() throws Exception {
+
+		configureAndStartMqttBroker();
+		
+		configureAasServer();
+		
+		startAasServer();
+
+		configureAndStartUpdaterComponent();
+		
+	}
+	
+	@AfterClass
+	public static void tearDown() {
+		updater.stopComponent();
+		
+		aasServer.stopComponent();
+	}
+	
+	@Test
+	public void getUpdatedPropertyValueA() throws MqttException, MqttSecurityException, MqttPersistenceException, InterruptedException, JsonProcessingException {
+		
+		String topicA = "aas/pressure";
+		String expectedValueA = "103.5585973";
+		
+		assertPropertyValueAorB(expectedValueA, topicA);
+	}
+	
+	@Test
+	public void getUpdatedPropertyValueB() throws MqttException, MqttSecurityException, MqttPersistenceException, InterruptedException, JsonProcessingException {
+
+		String topicB = "aas/rotation";
+		String expectedValueB = "379.5784558";
+		
+		assertPropertyValueAorB(expectedValueB, topicB);
+	}
+	
+	
+	@Test
+	public void getUpdatedPropertyValueC() throws MqttException, MqttSecurityException, MqttPersistenceException, InterruptedException, JsonProcessingException {
+
+		String topicB = "aas/pressure_rotation";
+		String expectedValueB = "[{\"modelType\":{\"name\":\"Property\"},\"kind\":\"Instance\",\"value\":\"103.5585973\",\"valueType\":\"anyType\",\"idShort\":\"pressure\",\"qualifiers\":[],\"semanticId\":{\"keys\":[]},\"parent\":{\"keys\":[{\"type\":\"Submodel\",\"local\":true,\"value\":\"https://example.com/ids/sm/8583_3140_7032_9766\",\"idType\":\"IRI\"}]}},{\"modelType\":{\"name\":\"Property\"},\"kind\":\"Instance\",\"value\":\"379.5784558\",\"valueType\":\"anyType\",\"idShort\":\"rotation\",\"qualifiers\":[],\"semanticId\":{\"keys\":[]},\"parent\":{\"keys\":[{\"type\":\"Submodel\",\"local\":true,\"value\":\"https://example.com/ids/sm/8583_3140_7032_9766\",\"idType\":\"IRI\"}]}}]";
+		
+		assertPropertyValueC(expectedValueB, topicB);
+	}
+	
+	private void assertPropertyValueAorB(String expectedValue, String topic) throws MqttSecurityException, MqttPersistenceException, MqttException, InterruptedException {
```

Comment:
Please rename this to assertPropertyValue

Reply:
changed